crates/stages/stages/src/sets.rs

```diff
@@ -286,8 +288,20 @@ where
         .add_set(HashingStages { stages_config: self.stages_config.clone() })
         .add_set(HistoryIndexingStages {
             stages_config: self.stages_config.clone(),
-            prune_modes: self.prune_modes,
-        })
+            prune_modes: self.prune_modes.clone(),
+        });
+
+        // If any prune modes are set, add the prune stage.
+        if !self.prune_modes.is_empty() {
+            // Prune stage should be added after all hashing stages, because otherwise it will
+            // delete
+            builder = builder.add_stage(PruneStage::new(
+                self.prune_modes.clone(),
+                self.stages_config.prune.commit_threshold,
+            ));
+        }
```

Comment:
let's add `fn add_stage_if` so we can keep using builder pattern

Reply:
done